protocol-units/movement-rest/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+use anyhow::Error;
+use aptos_api::Context;
+use poem::listener::TcpListener;
+use poem::{
+	get, handler,
+	middleware::{AddData, Tracing},
+	web::{Data, Path},
+	EndpointExt, IntoResponse, Response, Route, Server,
+};
+use std::env;
+use std::sync::Arc;
+use tracing::info;
+
+#[derive(Debug)]
+pub struct MovementRest {
+	/// The URL to bind the REST service to.
+	pub url: String,
+	pub context: Option<Arc<Context>>,
+	// More fields to be added here, log verboisty, etc.
+}
+
+impl MovementRest {
```

Comment:
This will obviously have to change at some point soonish to use the `config.toml`.